tests/stdlib/trandom.nim

```diff
@@ -37,23 +24,41 @@ proc main =
   doAssert sample("a") == 'a'
 
   when compileOption("rangeChecks"):
-    try:
+    doAssertRaises(RangeDefect):
       discard rand(-1)
-      doAssert false
-    except RangeDefect:
-      discard
 
-    try:
+    doAssertRaises(RangeDefect):
       discard rand(-1.0)
-      doAssert false
-    except RangeDefect:
-      discard
-
 
   # don't use causes integer overflow
   doAssert compiles(rand[int](low(int) .. high(int)))
 
-randomize(223)
 
-for i in 0 .. 10:
-  main()
+main()
+
+import math
+
+block:
+  doAssert almostEqual(rand(12.5), 4.012897747078944)
+  doAssert almostEqual(rand(2233.3322), 879.702755321298)
+
+  type DiceRoll = range[0..6]
+  doAssert rand(DiceRoll).int == 4
+
+var rs: RunningStat
+for j in 1..5:
+  for i in 1 .. 1_000:
+    rs.push(gauss())
+  doAssert abs(rs.mean-0) < 0.08, $rs.mean
+  doAssert abs(rs.standardDeviation()-1.0) < 0.1
+  let maxVal = 3.0
+  doAssert abs(rs.min + maxVal) < 1.0
+  doAssert abs(rs.max-maxVal) < 1.0
+  rs.clear()
+
+type DiceRoll = range[3..6]
+var flag = false
+for i in 0..<100000:
```

Comment:
optional: this test is valid (now that we're back to the saner `rand(max: int): int`) but maybe a 1 line comment would help the next reader; nit: also `100000` is overkill.